include/seqan3/io/detail/take_until_view.hpp

```diff
@@ -0,0 +1,684 @@
+
+// ============================================================================
+//  detail::take_until_or_throw (adaptor instance definition)
+// ============================================================================
+
+/*!\brief A view adaptor that returns elements from the underlying range until the functor evaluates to true
+ *        (**throws** if the end of the underlying range is reached).
+ * \throws seqan3::unexpected_end_of_input If the underlying range contains no element that satisfies the functor.
+ * \ingroup views
+ *
+ * \copydetails seqan3::detail::take_until
+ * \hideinitializer
+ */
+inline auto constexpr take_until_or_throw = take_until_fn<true, false>{};
+
+// ============================================================================
+//  detail::take_until_and_consume (adaptor instance definition)
+// ============================================================================
+
+/*!\brief A view adaptor that returns elements from the underlying range until the functor evaluates to true
+ *        (or the end of the underlying range is reached; consumes end in single-pass ranges).
+ * \throws seqan3::unexpected_end_of_input If the underlying range contains no element that satisfies the functor.
+ * \ingroup views
+ *
+ * \copydetails seqan3::detail::take_until
+ * \hideinitializer
+ */
+inline auto constexpr take_until_and_consume = take_until_fn<false, true>{};
+
+// ============================================================================
+//  detail::take_until_or_throw_and_consume (adaptor instance definition)
+// ============================================================================
+
+/*!\brief A view adaptor that returns elements from the underlying range until the functor evaluates to true
+ *        (**throws** if the end of the underlying range is reached; consumes end in single-pass ranges).
+ * \throws seqan3::unexpected_end_of_input If the underlying range contains no element that satisfies the functor.
+ * \ingroup views
+ *
+ * \copydetails seqan3::detail::take_until_and_consume
+ * \hideinitializer
+ */
+inline auto constexpr take_until_or_throw_and_consume = take_until_fn<true, true>{};
+
+//!\}
+
+} // namespace seqan3::detail
+
+namespace seqan3::views
+{
+
+/*!\brief A view adaptor that returns elements from the underlying range until the functor evaluates to
+ *        true (or the end of the underlying range is reached).
+ * \ingroup io
+ * \deprecated Use std::views::take_while with std::not_fn instead.
+ */
+SEQAN3_DEPRECATED_310 inline auto constexpr take_until = detail::take_until;
+
+/*!\brief A view adaptor that returns elements from the underlying range until the functor evaluates to true
+ *        (**throws** if the end of the underlying range is reached).
+ * \throws seqan3::unexpected_end_of_input If the underlying range contains no element that satisfies the functor.
+ * \ingroup io
+ * \deprecated Use std::views::take_while with std::not_fn.
```

Comment:
sorry I found one more
```suggestion
 * \deprecated Use std::views::take_while with std::not_fn instead.
```